### PR TITLE
JEP-200 - Cleanup the whitelist/blacklist terminology

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: jep/200/README.adoc [hunk 1]
```diff
@@ -429,3 +429,8 @@ BDFL confirms that the current test coverage is good enough to accept this Jenki
 * link:https://docs.google.com/document/d/1uQcyaaLvGFwFDe0mQ27JHeG2icdX0XfCHILbHGOtAmA/edit[JEP-200 Testing Notes (BDFL Review)]
 * link:https://github.com/jenkins-infra/jenkins.io/pull/1293[Announcement Blog Post Draft]
 * link:https://wiki.jenkins.io/display/JENKINS/Plugins+affected+by+fix+for+JEP-200[Wiki Page: Plugins affected by JEP-200]
+
+== Revision history
+
+* April 26, 2021 - Update the terminology in the JEP.
+  No other changes in the document.
```
Comment (jglick) [inline jep/200/README.adoc:436]:
```suggestion
```

`git blame`

**Review by jglick — COMMENTED** [review]
:shrug: -0, I guess; given that the source code uses the original terms, and I do not foresee that changing, this just seems likely to confuse readers.

File: jep/200/README.adoc (hunk 1 above)
Comment (oleg-nenashev) [reply jep/200/README.adoc:436]: git blame is not user friendly. And I do not want to offer commit history there, just note for major changes in the JEPs

**Comment by daniel-beck** [comment]
@oleg-nenashev Could you provide a reference for when/where these terms were decided on?

**Comment by oleg-nenashev** [comment]
Indeed it needs an agreement. Created https://docs.google.com/document/d/1-8myIWOZZktR0HNtbFIiNA0RfDCvkfKKuNI0C3wcvbo/edit#heading=h.i0px85o6agnq to track these agreements

File: jep/200/README.adoc [hunk 2]
```diff
@@ -67,52 +67,52 @@ Java’s built-in API, from the Remoting layer;
...
-* Classes mentioned in the original blacklist continue to be rejected.
+* Classes mentioned in the original denylist continue to be rejected.
 * Java arrays and enums are accepted, as these are special formats.
 * Java `Throwable` types are accepted, as they are commonly used in Remoting responses and it is impractical to enumerate every exception that might be thrown.
 * A class defined in Jenkins core, Remoting, or a Jenkins module/plugin is accepted. This does _not_ apply to classes defined in other JARs packaged in `WEB-INF/lib/*.jar`.
 * Some classes defined in test utilities, by mocking frameworks, etc., are accepted when running inside tests.
 * Classes defined in a `RemoteClassLoader` (so, already sent to this JVM from a presumably trusted source) are accepted.
   This is needed at least by CloudBees Jenkins Operations Center, and clients of the `master-to-master-api` plugin if there are any.
-* Any other class is accepted if it is explicitly mentioned in a list `whitelisted-classes.txt` bundled in Jenkins core. These consist of:
+* Any other class is accepted if it is explicitly mentioned in a list `permitlisted-classes.txt` bundled in Jenkins core. These consist of:
```
Comment (jglick) [inline jep/200/README.adoc:92]: revert

**Comment by oleg-nenashev** [comment]
Will need an update, because the consensus is to use allowlist so far

**Review by jglick — COMMENTED** [review]
as you prefer

File: jep/200/README.adoc [hunk 3]
```diff
@@ -1,4 +1,4 @@
-= JEP-200: Switch Remoting/XStream blacklist to a whitelist
+= JEP-200: Switch Remoting/XStream denylist to a allowlist
```
Comment (daniel-beck) [inline jep/200/README.adoc:1]:
```suggestion
= JEP-200: Switch Remoting/XStream denylist to an allowlist
```

File: jep/200/README.adoc [hunk 4]
```diff
@@ -16,7 +16,7 @@ endif::[]
 | 200
 
 | Title
-| Switch Remoting/XStream blacklist to a whitelist
+| Switch Remoting/XStream denylist to a allowlist
```
Comment (daniel-beck) [inline jep/200/README.adoc:19]:
```suggestion
| Switch Remoting/XStream denylist to an allowlist
```

File: jep/200/README.adoc [hunk 5]
```diff
@@ -65,24 +65,24 @@ endif::[]
 Since its early days, Jenkins has used two object serialization frameworks:
 Java’s built-in API, from the Remoting layer;
 and XStream, for persisting configuration and settings.
-Both permit any Java class to be loaded merely by specifying its name in the input.
+Both allow any Java class to be loaded merely by specifying its name in the input.
 
-This proposal switches the blacklist to a whitelist of a few dozen entries:
+This proposal switches the denylist to a allowlist of a few dozen entries:
```
Comment (daniel-beck) [inline jep/200/README.adoc:70]:
```suggestion
This proposal switches the denylist to an allowlist of a few dozen entries:
```

File: jep/200/README.adoc [hunk 6]
```diff
@@ -196,14 +196,14 @@ so the maintainers will be the entry point.
 
 For years, the Jenkins project has received reports of remote code execution (RCE) attacks involving these frameworks.
 Typically the attacks involve fairly exotic classes in the Java Platform, or sundry libraries such as Groovy.
-The Jenkins CERT has responded to such reports reactively, by blacklisting the affected classes or packages.
+The Jenkins CERT has responded to such reports reactively, by prohibiting the affected classes or packages.
 That approach has proven unmaintainable, as there is a constant threat of further exploits using unexamined classes.
 
-This proposal switches the blacklist to a whitelist of a few dozen entries.
+This proposal switches the denylist to a allowlist of a few dozen entries.
```
Comment (daniel-beck) [inline jep/200/README.adoc:202]:
```suggestion
This proposal switches the denylist to an allowlist of a few dozen entries.
```

File: jep/200/README.adoc [hunk 7]
```diff
@@ -234,26 +234,26 @@ Similarly, after 2017-02-01 a potential attack vector involving console notes (m
...
 (Note: Pipeline builds based on the Groovy CPS engine use yet another serialization framework, JBoss Marshalling, to save state.
 This is not considered a security issue since the `program.dat` files are never read from user data.)
 
 == Reasoning
 
-The CERT team could continue to expand the blacklist in response to newly reported vulnerabilities.
+The CERT team could continue to expand the denylist in response to newly reported vulnerabilities.
 This has proven to be a significant maintenance burden, and there is little trust in the result.
-Outside security authorities have repeatedly urged the Jenkins team to switch to a whitelist.
+Outside security authorities have repeatedly urged the Jenkins team to switch to a allowlist.
```
Comment (daniel-beck) [inline jep/200/README.adoc:256]:
```suggestion
Outside security authorities have repeatedly urged the Jenkins team to switch to an allowlist.
```

File: jep/200/README.adoc [hunk 8]
```diff
@@ -326,25 +326,25 @@ Some runs of `acceptance-test-harness` (ATH) were already performed in this mode
...
-If new whitelist entries are needed after release, they can be added to core in weekly updates.
-Plugins can also contribute their own whitelist (or even blacklist) entries for third-party libraries they bundle,
+If new allowlist entries are needed after release, they can be added to core in weekly updates.
+Plugins can also contribute their own allowlist (or even denylist) entries for third-party libraries they bundle,
 as described in Extensibility above.
 
-Finally, an individual administrator can define site-specific whitelist (or blacklist) entries with a system property `hudson.remoting.ClassFilter`.
-This could be useful as an emergency measure, permitting functionality to be restored while awaiting a new plugin release.
+Finally, an individual administrator can define site-specific allowlist (or denylist) entries with a system property `hudson.remoting.ClassFilter`.
+This could be useful as an emergency measure, allowing functionality to be restored while awaiting a new plugin release.
```
Comment (daniel-beck) [inline jep/200/README.adoc:334]: why?

File: jep/200/README.adoc [hunk 9]
```diff
@@ -326,25 +326,25 @@ Some runs of `acceptance-test-harness` (ATH) were already performed in this mode
...
 
 == Security
 
 This proposal is expected to strictly improve Jenkins security,
-as the existing blacklist is retained as a fallback unless deliberately overridden.
+as the existing denylist is retained as a fallback unless deliberately overridden.
 
 == Infrastructure Requirements
 
 A new redirect `https://jenkins.io/redirect/class-filter/` has been offered, pointing to a wiki page.
```
Comment (daniel-beck) [inline jep/200/README.adoc:346]:
```suggestion
A new redirect `https://jenkins.io/redirect/class-filter/` has been offered, pointing to documentation for this feature.
```

File: jep/200/README.adoc [hunk 10]
```diff
@@ -369,18 +369,18 @@ was already detected by an automated test run, and a simple fix proposed and mer
...
 link:https://github.com/jenkinsci/git-plugin/pull/497[PR 497]
 the `git` plugin does a lot of tricky things with the Eclipse JGit library.
 That is true even if you have specified the CLI implementation of Git for use in the build!
 In this case, `GitSCM.printCommitMessageToLog` asks the agent to return a `RevCommit` (a JGit type),
-which is serialized and deserialized, and then the master calls `getShortMessage()` on that structure.
+which is serialized and deserialized, and then the controller calls `getShortMessage()` on that structure.
 It would be simpler, faster, and safer to do this processing on the agent and send back a `String`,
 but the deceptive ease of Remoting tempts developers to do the wrong thing.
-Enforcing a whitelist in the baseline version of Jenkins might help guide them to the simpler solution.
+Enforcing a allowlist in the baseline version of Jenkins might help guide them to the simpler solution.
```
Comment (daniel-beck) [inline jep/200/README.adoc:383]:
```suggestion
Enforcing an allowlist in the baseline version of Jenkins might help guide them to the simpler solution.
```

File: jep/200/README.adoc (hunk 1 above)
Comment (daniel-beck) [reply jep/200/README.adoc:436]: Using a different term to refer to the same thing with no functional impact is not a "major change".

File: jep/README.adoc [hunk 11]
```diff
@@ -83,7 +83,7 @@
 | TBD
 
 | Final{nbsp}:lock:
-| link:200/README.adoc[JEP-200: Switch Remoting/XStream blacklist to a whitelist]
+| link:200/README.adoc[JEP-200: Switch Remoting/XStream denylist to a permitlist]
```
Comment (daniel-beck) [inline jep/README.adoc:86]:
```suggestion
| link:200/README.adoc[JEP-200: Switch Remoting/XStream denylist to an allowlist]
```

**Review by daniel-beck — CHANGES_REQUESTED** [review]

File: jep/200/README.adoc (hunk 8 above)
Comment (oleg-nenashev) [reply jep/200/README.adoc:334]: Just fell into my search and I though "allowed" would sound better. Unrelated to the terminology cleanup

**Review by timja — APPROVED** [review]

**Pull request merged**